doc/developer-guide/documentation/RPC.en.rst

```diff
@@ -0,0 +1,397 @@
+.. Licensed to the Apache Software Foundation (ASF) under one
+   or more contributor license agreements.  See the NOTICE file
+   distributed with this work for additional information
+   regarding copyright ownership.  The ASF licenses this file
+   to you under the Apache License, Version 2.0 (the
+   "License"); you may not use this file except in compliance
+   with the License.  You may obtain a copy of the License at
+
+   http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing,
+   software distributed under the License is distributed on an
+   "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+   KIND, either express or implied.  See the License for the
+   specific language governing permissions and limitations
+   under the License.
+
+.. include:: ../../common.defs
+
+.. highlight:: cpp
+.. default-domain:: cpp
+
+.. _developer-doc-rpc:
+
+RPC
+*************
+
+========
+Overview
+========
+
+To communicate between :program:`traffic_manager` and :program:`traffic_server` there is an RPC mechanism in
+``mgmt/``. This is a simple serialization style RPC which runs over sockets.
+
+=======================================================
+Runtime Structure :program:`traffic_ctl` to :program:`traffic_server`
+=======================================================
+
+.. uml::
+   :align: center
+
+   hide empty members
+
+   node "traffic_manager"
+   node "traffic_server"
+
+   [traffic_ctl] <-d-> [traffic_manager] : Remote RPC
+   [traffic_cop] <-r-> [traffic_manager] : Remote RPC
+   [other remote clients] <-u-> [traffic_manager] : Remote RPC
+   [traffic_manager] <-r-> [traffic_server] : Local RPC
+   [traffic_server] <-r-> [plugin]
+
+:program:`traffic_manager` opens a socket to recieve commands from remote clients. :program:`traffic_manager` has a socket connection to :program:`traffic_server` to relay RPC commands.
+
+===========================================================
+Sequence Diagram from :program:`traffic_ctl` to :program:`traffic_server`
+===========================================================
+
+Sequence diagram for a command sent from :program:`traffic_ctl` to when it is recieved by a plugin. 
+
+.. image:: ../../uml/images/RPC-sequence-diagram.svg
+    
+.. note::
+
+    Currently a fire and forget model. traffic_manager sends out an asynchoronous signal without any acknowledgement. It then proceeds to send a response itself. 
+
+=======================
+Remote RPC vs Local RPC 
+=======================
+
+The RPC API for remote clients, such as :program:`traffic_ctl`, :program:`traffic_cop`, etc, is different from the RPC API used between :program:`traffic_manager` and :program:`traffic_server`. 
```

Comment:
You might look at local substitutions for these programs, as is done in `common.defs` but just in this file locally.

Reply:
updated.